torch/csrc/jit/codegen/onednn/kernel.cpp

```diff
@@ -0,0 +1,243 @@
+    auto constantInputSpecIdx = nGraphInputs_ + i;
+    auto constantInputSpec = inputSpecs_[constantInputSpecIdx];
+    runInputs.push_back({constantInputSpec.logical_tensor(),
+                         Engine::getEngine(),
+                         constantInputs_[i].data_ptr()});
+  }
+
+  for (size_t i = 0; i < nOutputs_; i++) {
+    auto spec = outputSpecs_[i];
+    auto opt = c10::TensorOptions(spec.aten_scalar_type()).device(device_);
+
+    auto outputId = spec.tid();
+    auto iter = inplacePairs_.find(outputId);
+    if (iter != inplacePairs_.end()) {
+      // output reuses one of input tensors
+      auto inputOffset = iter->second;
+      auto inputTensor = inputs[inputOffset];
+      outputs.push_back(inputTensor);
+      runOutputs.push_back(
+          {spec.logical_tensor(), Engine::getEngine(), inputTensor.data_ptr()});
+    } else if (spec.is_opaque()) {
+        auto tensor = empty_llga(spec, opt);
+        outputs.push_back(tensor);
+        runOutputs.push_back(llga_from_aten_tensor(tensor));
+    } else {
+      auto tensor = at::empty_strided(spec.sizes(), spec.strides(), opt);
+      outputs.push_back(tensor);
+      runOutputs.push_back(
+          {spec.logical_tensor(), Engine::getEngine(), tensor.data_ptr()});
+    }
+  }
+
+  return std::make_tuple(runInputs, runOutputs);
+}
+
+compiled_partition LlgaKernel::compile(const partition& partition) {
+  auto inputs = fmap(inputSpecs_, toLogicalTensor);
+  auto outputs = fmap(outputSpecs_, toLogicalTensor);
+  auto compilation = partition.compile(inputs, outputs, Engine::getEngine());
+
+  // Since layouts of opaque outputs would be known after compilation,
+  // we need to query them out from compilation and update outputSpecs
+  for (size_t i = 0; i < nOutputs_; i++) {
+    auto tid = outputSpecs_[i].tid();
+    outputSpecs_[i] = compilation.query_logical_tensor(tid);
+  }
+
+  // Build static mapping from output id to input offset
+  // in accordance with available inplace options
+  for (auto&& option : compilation.get_inplace_ports()) {
+    size_t inputId = option.first;
+    size_t outputId = option.second;
+    auto inputSpecIter =
+        std::find_if(inputSpecs_.begin(), inputSpecs_.end(), [&](auto& spec) {
+          return spec.tid() == inputId;
+        });
+    TORCH_CHECK(inputSpecIter != inputSpecs_.end(), "In-place input not found");
+    auto inputOffset = inputSpecIter - inputSpecs_.begin();
+    inplacePairs_[outputId] = inputOffset;
+  }
+
+  return compilation;
+}
+
+void LlgaKernel::run(Stack& stack) {
+  GRAPH_DEBUG("In ", debugName(), "\n");
+
+  // Grab input values from stack
+  auto stackInputs = last(stack, nGraphInputs_);
+  auto inputs = fmap(stackInputs, [&](const IValue& v) {
+    TORCH_CHECK(
+        v.isTensor(), "Stack values for LLGA partition must be Tensor type");
+    return v.toTensor();
+  });
+
+  // Even in case of concurrent threads, the kernel would be initialized once.
+  std::call_once(initialized_flag,
```

Comment:
If input tensor characteristics would somehow change, then they'd deviate from their corresponding oneDNN Graph's logical tensor, and the partition compilation would fail, resulting in an exception being thrown by oneDNN Graph.

Can you please provide an example of when input tensors might change? I'm guessing you might've encountered such an example. I'll try searching the commit history in the master branch to find such an example. Thanks!